src/snapred/backend/recipe/algorithm/ReductionAlgorithm.py

```diff
@@ -185,14 +188,21 @@ def PyExec(self):
         # TODO: Refactor so excute only needs to be called once
         self.mantidSnapper.executeQueue()
 
+        dMinList = []
+        dMaxList = []
+        dBinList = []
+        for i in pixelGroupingParameters:
+            dMinList.append(i.dResolution.minimum)
+            dMaxList.append(i.dResolution.maximum)
+            dBinList.append(i.dRelativeResolution)
         groupedData = data
         for workspaceIndex in range(len(focusGroups)):
             data = self.mantidSnapper.RebinRagged(
                 "Rebinning ragged bins...",
                 InputWorkspace=mtd[groupedData].getItem(workspaceIndex),
-                XMin=focusGroups[workspaceIndex].dMin,
-                XMax=focusGroups[workspaceIndex].dMax,
-                Delta=focusGroups[workspaceIndex].dBin,
+                XMin=dMinList[workspaceIndex],
+                XMax=dMaxList[workspaceIndex],
+                Delta=dBinList[workspaceIndex] / reductionIngredients.reductionState.instrumentConfig.NBins,
```

Comment:
Make this a loop over group ID in the (sorted) list of group IDs, and use the group ID as the key (after the above change)